apps/blog/src/posts/macos-pwa-install/macos-pwa-install.md

```diff
@@ -0,0 +1,88 @@
+
+![PWA Install Instructions](/posts/macos-pwa-install/pic-3.jpg)
+
+Why easy installation matters? Progressive Web Applications offer a lot of flexibility and user engagement. One of the main selling points is their app-like presence on your device. However, not every browser has made it easy to install them. This update is set to make the installation of PWAs feel more native, especially for users on macOS Safari.
+
+### Implementation
+
+But how it was archived? Safari on macOS Sonoma supports the Web App Manifest, which is a JSON file that describes the Web App metadata. This includes the app's name, description, icons, and more. The pwa-install component uses this information to create a native-like installation experience for PWAs on iOS/iPadOS and macOS Sonoma.
+
+But here's a catch: Safari on macOS don't provide any API to detect Web Apps availability. So we had to use a workaround to detect Web Apps availability on macOS platform.
+
+[Here's how it works](https://github.com/khmyznikov/pwa-install/blob/cf73d0c382fd87aa6b5a5cc40f0474150efe3487/src/utils.ts#L24):
+
+```js
+static isAppleDesktop(): boolean {
+    // check if it's a mac
+    const userAgent = navigator.userAgent.toLowerCase();
+    if (navigator.maxTouchPoints || !userAgent.match(/macintosh/))
+        return false;
+    // check safari version >= 17
+    const version = /version\/(\d{2})\./.exec(userAgent)
+    if (!version || !version[1] || !(parseInt(version[1]) >= 17))
+        return false;
+    // hacky way to detect Sonoma
+    const audioCheck = document.createElement('audio').canPlayType('audio/wav; codecs="1"') ? true : false;
+    const webGLCheck = new OffscreenCanvas(1, 1).getContext('webgl') ? true : false;
+
+    return audioCheck && webGLCheck;
+}
+```
+Combined the user agent and specific for Safari on Sonoma feature detections, we can detect Web Apps availability on macOS platform. We had to run previous and current macOS version with latest Safari 17 side by side and with help of [tool like this](https://browserleaks.com/features) we were able to detect the difference we can use.
+
+### Installation
+
+This isn't just about Safari on MacOS. The component keeps the user experience consistent no matter the device.
+
+| &nbsp;&nbsp;iOS&nbsp;&nbsp; | &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; | Android |
+| --- | --- | --- |
+|![iOS install prompt](/posts/macos-pwa-install/pic-4.jpg)|![iOS install instructions](/posts/macos-pwa-install/pic-5.jpg)|![Android Gallery](/posts/macos-pwa-install/pic-6.jpg)|
+
+![Chrome Prompt](/posts/macos-pwa-install/pic-7.jpg)
```

Comment:
Should we show the prompt on Edge instead of on Chrome?

Reply:
Hehe nice catch